packages/ses/src/tame-function-constructors.js

```diff
@@ -45,11 +50,40 @@ import {
 export default function tameFunctionConstructors() {
   try {
     // Verify that the method is not callable.
-    // eslint-disable-next-line @endo/no-polymorphic-call
-    FERAL_FUNCTION.prototype.constructor('return 1');
-  } catch (ignore) {
+    FERAL_EVAL('');
+  } catch {
+    const unsafeFunctionConstructors = arrayFilter(
+      arrayMap(
+        [
+          function unsafeFunction() {},
+          // eslint-disable-next-line no-empty-function
+          function* unsafeGeneratorFunction() {},
+          // eslint-disable-next-line no-empty-function
+          async function unsafeAsyncFunction() {},
+          // eslint-disable-next-line no-empty-function
+          async function* unsafeAsyncGeneratorFunction() {},
+        ],
+        f => f.constructor,
+      ),
+      constructor => {
+        try {
+          constructor('');
```

Comment:
I may be paranoid, but wouldn't there technically be a way to bypass this by allowing the "empty" / null sha in CSP?

I guess it'd really be the embedder shooting itself in the foot in that case.